app/controllers/detailedvaluationrequest/DvrController.scala

```diff
@@ -281,19 +298,62 @@ class DvrController @Inject()(
           .url
     }
   }
+
+  def canChallenge(
+        plSubmissionId: String,
+        assessmnetRef: Long,
+        caseRef: String,
+        authorisationId: Long,
+        uarn: Long,
+        isOwner: Boolean): Action[AnyContent] = authenticated.async { implicit request =>
+    propertyLinks.canChallenge(plSubmissionId, assessmnetRef, caseRef, isOwner) flatMap {
+      case None =>
+        Future.successful(Redirect(config.businessRatesValuationFrontendUrl(
+          s"/property-link/$plSubmissionId/assessment/$assessmnetRef/startChallenge?propertyLinkId=$authorisationId&isOwner=$isOwner")))
+      case Some(response) => {
+        response.result match {
+          case true => {
+            val party = if (isOwner) "client" else "agent"
```

Comment:
do we reference these hard-coded strings in many places?

Reply:
nope, they are only used here